File: js-bindings/examples/async.js
/**
 * Example of sending an email asynchronously with micromail
 */

const micromail = require('../index');

// Create a configuration
const config = micromail.createConfig("example.com");

// Create a mailer
const mailer = micromail.createMailer(config);

// Create an email
const mail = micromail.createMail();
micromail.mailSetFrom(mail, "sender@example.com");
micromail.mailSetTo(mail, "recipient@example.com");
micromail.mailSetSubject(mail, "Hello from Node.js (async)");
micromail.mailSetBody(mail, "This is a test email sent from Node.js using async/await!");

// Send the email asynchronously
console.log("Sending email asynchronously...");

micromail.mailerSendAsync(mailer, mail)
    .then(() => {
        console.log("Email sent successfully!");
        
        // Get the log
        const log = micromail.mailerGetLog(mailer);
        console.log("\nLog:");
        log.forEach(line => console.log(line));
    })
    .catch(err => {
        console.error("Failed to send email:", err);
    });

File: examples/async.rs
//! Example of sending an email with async/await

use micromail::{Config, AsyncMailer, async_mail::AsyncMailSender, Mail, Error};

#[tokio::main]
async fn main() -> Result<(), Error> {
    
    // Create a configuration
    let config = Config::new("example.com");
    
    // Create an async mailer
    let mut mailer = AsyncMailer::new(config);
    
    // Create an email
    let mail = Mail::new()
        .from("sender@example.com")
        .to("recipient@example.com")
        .subject("Hello from micromail (async)")
        .body("This is a test email sent with micromail using async/await!");
    
    // Send the email asynchronously
    match mailer.send(mail).await {
        Ok(_) => {
            println!("Email sent successfully!");
            
            // Get the inner mailer to access the log
            let inner = mailer.mailer();
            let guard = inner.lock().unwrap();
            
            // Print the log
            println!("\nLog:");
            for line in guard.get_log() {
                println!("{}", line);
            }
        }
        Err(e) => {
            println!("Failed to send email: {}", e);
        }
    }
    
    Ok(())
}